### PR TITLE
[TimeSeriesInsights] Updates to Hierarchy API and take parameter in GetEvents, GetSeries and progress in all query APIs

### PR DESCRIPTION
**Pull request by yeskarthik**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json [hunk 1]
```diff
@@ -1777,58 +1820,248 @@
...
+          }
+        }
+      }
+    },
+    "HierarchiesSortParameter": {
+      "description": "Definition of sorting of hierarchy nodes.",
+      "type": "object",
+      "properties": {
+        "by": {
+          "description": "Value to use for hierarchy node sorting. When it is set to 'CumulativeInstanceCount', the returned hierarchies are sorted based on the total instances belonging to the hierachy node and its child hierarchy nodes. When it is set to 'Name', the returned hierarchies are sorted based on the hierarchy name. Optional, default is 'CumulativeInstanceCount'.",
```
Comment (nschonni) [inline specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]: hierachy -> hierarchy
```suggestion
          "description": "Value to use for hierarchy node sorting. When it is set to 'CumulativeInstanceCount', the returned hierarchies are sorted based on the total instances belonging to the hierarchy node and its child hierarchy nodes. When it is set to 'Name', the returned hierarchies are sorted based on the hierarchy name. Optional, default is 'CumulativeInstanceCount'.",
```

File: specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json [hunk 2]
```diff
@@ -1872,6 +2113,29 @@
...
+      "description": "The hierarchy node which contains the instances matching the query based on the input. May be empty or null.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the hierarchy node. May be empty, cannot be null.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "cumulativeInstanceCount": {
+          "description": "Total number of instances that belong to this node and it's subtrees matching the query.",
```
Comment (nschonni) [inline specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]: You can add "subtrees" to https://github.com/Azure/azure-rest-api-specs/blob/master/custom-words.txt if it's the right word

Comment (yeskarthik) [reply specification/timeseriesinsights/data-plane/Microsoft.TimeSeriesInsights/preview/2018-11-01-preview/timeseriesinsights.json]: I have added it in the branch.

**Comment by nschonni** [comment]
The errors in Avocado for `invalid symbol, token: \uFEFF, line: 1, column: 1` are for JSON files with BOMs. You'll need to resave the as UTF-8 without BOM to fix them

**Review by tjprescott — APPROVED** [review]

**Pull request merged**